src/nvim/move.c

```diff
@@ -506,6 +508,14 @@ void check_cursor_moved(win_T *wp)
   if (wp->w_cursor.lnum != wp->w_valid_cursor.lnum) {
     wp->w_valid &= ~(VALID_WROW|VALID_WCOL|VALID_VIRTCOL
                      |VALID_CHEIGHT|VALID_CROW|VALID_TOPLINE);
+
+    // Concealed line visibility toggled.
+    if (wp->w_p_cole >= 2 && !conceal_cursor_line(wp)
+        && (decor_conceal_line(wp, wp->w_buffer, wp->w_cursor.lnum - 1)
+            || decor_conceal_line(wp, wp->w_buffer, wp->w_valid_cursor.lnum - 1))) {
+      // TODO(luukvbaal): shouldn't have to redraw entire window but scroll instead.
+      wp->w_lines_valid = 0;
+    }
```

Comment:
Does this work if you move it to `redraw_for_cursorline()`?

Reply:
Unfortunately it doesn't. I checked that before and debugged why it doesn't but forgot x)